src/bitcoin/electrum/client.rs

```diff
@@ -0,0 +1,375 @@
+                    .expect("all required txs are in graph");
+                anc_fees += fee;
+                anc_size += anc_tx.vsize();
+            } else {
+                log::debug!("Ancestor txid '{}' is not unconfirmed.", anc_tx.txid());
+                continue;
+            }
+        }
+        let fees = MempoolEntryFees {
+            base: bitcoin::Amount::from_sat(base_fee),
+            ancestor: bitcoin::Amount::from_sat(anc_fees),
+            descendant: bitcoin::Amount::from_sat(desc_fees),
+        };
+        let entry = MempoolEntry {
+            vsize: base_size.try_into().expect("tx size must fit into u64"),
+            fees,
+            ancestor_vsize: anc_size.try_into().expect("tx size must fit into u64"),
+        };
+        // It's possible that the chain tip has now changed, but it hadn't done as of the last sync,
+        // so go ahead and return the results.
+        Ok(Some(entry))
+    }
+
+    /// Get mempool spenders of the given outpoints.
+    ///
+    /// Will restart if chain tip changes before completion.
+    pub fn mempool_spenders(
+        &self,
+        outpoints: &[bitcoin::OutPoint],
+    ) -> Result<Vec<MempoolEntry>, Error> {
+        log::debug!("Getting mempool spenders for outpoints: {:?}.", outpoints);
+        const BATCH_SIZE: usize = 200;
+        let mut local_chain = LocalChain::from_genesis_hash(self.genesis_block().hash).0;
+        let chain_tip = self.chain_tip();
+        if chain_tip.height > 0 {
+            let _ = local_chain
+                .insert_block(block_id_from_tip(chain_tip))
+                .expect("only contains genesis block");
+        }
+        let local_tip = local_chain.tip();
+        let request =
+            SyncRequest::from_chain_tip(local_chain.tip()).chain_outpoints(outpoints.to_vec());
+        // We don't need to fetch prev txouts as we just want the outspends.
+        let sync_result = self.sync_with_confirmation_height_anchor(request, BATCH_SIZE, false)?;
+        let _ = local_chain.apply_update(sync_result.chain_update);
+        if local_chain.tip() != local_tip {
+            log::debug!("Chain tip changed while getting mempool spenders. Restarting.");
+            return self.mempool_spenders(outpoints);
+        }
+        let graph = sync_result.graph_update;
+        let txids: HashSet<_> = outpoints
+            .iter()
+            .flat_map(|op| graph.outspends(*op))
+            .collect();
+        let mut entries = Vec::new();
+        for txid in txids {
+            let entry = self.mempool_entry(txid)?;
+            if let Some(entry) = entry {
+                entries.push(entry);
+            }
+        }
+        // Make sure the tip didn't change while going through each of the txids.
+        if self.chain_tip() != chain_tip {
+            log::debug!("Chain tip changed while getting mempool spenders. Restarting.");
+            return self.mempool_spenders(outpoints);
+        }
```

Comment:
I changed it so that the `mempool_spenders` function can specify the tip it's expecting when calling `mempool_entries` and so I could remove this final check.